libp2p/NodeTable.cpp

```diff
@@ -396,49 +383,24 @@ NodeTable::NodeBucket& NodeTable::bucket_UNSAFE(NodeEntry const* _n)
 
 void NodeTable::onReceived(UDPSocketFace*, bi::udp::endpoint const& _from, bytesConstRef _packet)
 {
-	// h256 + Signature + type + RLP (smallest possible packet is empty neighbours packet which is 3 bytes)
-	if (_packet.size() < h256::size + Signature::size + 1 + 3)
-	{
-		clog(NodeTableTriviaSummary) << "Invalid message size from " << _from.address().to_string() << ":" << _from.port();
-		return;
-	}
-	
-	bytesConstRef hashedBytes(_packet.cropped(h256::size, _packet.size() - h256::size));
-	h256 hashSigned(sha3(hashedBytes));
-	if (!_packet.cropped(0, h256::size).contentsEqual(hashSigned.asBytes()))
-	{
-		clog(NodeTableTriviaSummary) << "Invalid message hash from " << _from.address().to_string() << ":" << _from.port();
-		return;
-	}
-	
-	bytesConstRef signedBytes(hashedBytes.cropped(Signature::size, hashedBytes.size() - Signature::size));
-
-	// todo: verify sig via known-nodeid and MDC
-	
-	bytesConstRef sigBytes(_packet.cropped(h256::size, Signature::size));
-	Public nodeid(dev::recover(*(Signature const*)sigBytes.data(), sha3(signedBytes)));
-	if (!nodeid)
-	{
-		clog(NodeTableTriviaSummary) << "Invalid message signature from " << _from.address().to_string() << ":" << _from.port();
-		return;
-	}
-	
-	unsigned packetType = signedBytes[0];
-	bytesConstRef rlpBytes(_packet.cropped(h256::size + Signature::size + 1));
 	try {
-		RLP rlp(rlpBytes);
-		switch (packetType)
+		auto packet = DiscoveryDatagram::interpretUDP(_from, _packet);
```

Comment:
Please remove DRY from this PR and submit in a future PR, or, leave DRY in and submit functional changes in a separate PR.